src/Altinn.Apps/AppFrontend/react/altinn-app-frontend/src/components/advanced/AddressComponent.tsx

```diff
@@ -53,16 +54,8 @@ export function AddressComponent(props: IAddressComponentProps) {
   const [validations, setValidations] = React.useState({} as any);
 
   React.useEffect(() => {
-    // eslint-disable-next-line @typescript-eslint/no-shadow
-    const zipCode = props.formData.zipCode;
-    if (!zipCode || zipCode === '') {
-      return;
-    }
-
-    if (!zipCode.match(new RegExp('^[0-9]{4}$'))) {
-      const errorMessage = getLanguageFromKey('address_component.validation_error_zipcode', props.language);
+    if (!zipCode || zipCode === '' || !zipCode.match(new RegExp('^[0-9]{4}$'))) {
```

Comment:
`!zipCode` and `zipCode === ''` will both evaluate to the true if string is empty, can remove explicit empty string check